demos/mqtt/mqtt_demo_mutual_auth/CMakeLists.txt

```diff
@@ -35,20 +35,21 @@ target_include_directories(
         ${LOGGING_INCLUDE_DIRS}
 )
 
-# Download the Amazon Root CA certificate.
-message( "Downloading the Amazon Root CA certificate..." )
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/certificates)
-execute_process(
-    COMMAND curl --url https://www.amazontrust.com/repository/AmazonRootCA1.pem
-    -o ${CMAKE_CURRENT_LIST_DIR}/certificates/AmazonRootCA1.crt
-)
-
-# Copy the certificates and client key to the binary directory.
-add_custom_command(
-    TARGET
-        ${DEMO_NAME}
-    POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy_directory
-        "${CMAKE_CURRENT_LIST_DIR}/certificates"
-        "$<TARGET_FILE_DIR:${DEMO_NAME}>/certificates"
-)
+if(ROOT_CA_CERT_PATH)
+    target_compile_definitions(
+        ${DEMO_NAME} PRIVATE
+            ROOT_CA_CERT_PATH="${ROOT_CA_CERT_PATH}"
```

Comment:
Even if the file path is passed as a cmake flag, will it still work properly from any directory if the given flag is a relative path and not an absolute path? If not, is it possible to parse this flag using `get_filename_component` and convert it into an absolute path first?

Reply:
Good idea!